firedrake/assemble.py

```diff
@@ -20,7 +20,7 @@
 
 
 def assemble(f, tensor=None, bcs=None, form_compiler_parameters=None,
-             inverse=False, nest=None):
+             inverse=False, nest=None, matfree=False, extra_ctx={}):
```

Comment:
But it's tough to give a default value to mean "the way Firedrake is on
master" if we have "monolithic" for 1-forms.  With some slight abuse of
notation, I'd like to
have "aij" for vector assembly...

On Mon, Aug 15, 2016 at 1:45 PM, Robert Kirby <robert.c.kirby@gmail.com>
wrote:

> Ok, we'll patch that up later, but before the merge, then.
>
> Next question (this is in flight, but messy).  I just replaced the "nest"
> + "matfree" options with "mat_type" in the argument list.  But what about
> 1-forms and 0-forms.
> Vectors could be monolithic or nested.  I guess they could be implicit,
> but can't really think of a Firedrake use case.
>
> Maybe "assembly_type" instead of "mat_type" is a better keyword.  Then
> "assembly_type" can be "aij", "nest", or "matfree" if we have a 2-form and
> "monolithic" or "nest" if a 1-form.
>
> That's a slight asymmetry between 1-forms and 2-forms, but preserves the
> possibility of future assembled matrix types like MATIS.
>
> Thoughts?
>
>
>
> On Mon, Aug 15, 2016 at 11:17 AM, Lawrence Mitchell <
> notifications@github.com> wrote:
>
>> In firedrake/assemble.py
>> <https://github.com/firedrakeproject/firedrake/pull/853#discussion_r74787473>
>> :
>>
>> > @@ -20,7 +20,7 @@
>> >
>> >
>> >  def assemble(f, tensor=None, bcs=None, form_compiler_parameters=None,
>> > -             inverse=False, nest=None):
>> > +             inverse=False, nest=None, matfree=False, extra_ctx={}):
>>
>> I'll add backwards compat and deprecation warning after the changes.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/firedrakeproject/firedrake/pull/853/files/ff5ec4e0eabb94576ac69c9c82932eaa50b2d14e#r74787473>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ADl6L6W7BJR7m8OrH2qlCT0bd5g47JEOks5qgJEJgaJpZM4JiV-g>
>> .
>>
>
>